stream_alert/rule_processor/parsers.py

```diff
@@ -104,19 +104,28 @@ def _key_check(self, json_records):
         """
         schema = self._get_schema()
         schema_keys = set(schema.keys())
+        valid_records = []
+        schema_match = False
```

Comment:
Since you're appending to a new list now, I think you can simplify this loop and remove the boolean that tracks the validity. Like:

```
        valid_records = []

        for json_record in json_records:
            json_keys = set(json_record.keys())
            if json_keys == schema_keys:
                for key, key_type in schema.iteritems():
                    # If the value is a map of defined key/value pairs
                    if isinstance(key_type, dict) and key_type:
                        # subkey check
                        if set(json_record[key].keys()) == set(schema[key].keys()):
                            valid_records.append(json_record)

            else:
                logger.debug('JSON Key mismatch: %s vs. %s', json_keys, schema_keys)

        return valid_records
```

Also note change for the check for the empty dict (`and key_type` from `and key_type != {}`). This same sort of logic applies to empty lists. len(dict) will also do the same. You can take that change or leave it...